Asset Uploader UI Component/Code/Asset Uploader/Objective-C/AssetUploader/ViewController.m

```diff
@@ -0,0 +1,298 @@
+        {
+            AdobeLibraryComposite *composite = destination.selectedItem;
+            NSError *error;
+            
+            // Add assets to selected library and perform sync.
+            [AdobeDesignLibraryUtils addImage:assetURL
+                                         name:assetName
+                                      library:composite
+                                        error:&error];
+            
+            if (!error)
+            {
+                NSLog(@"Added to library: %@", assetName);
+            }
+            else
+            {
+                NSLog(@"Add to library failed: %@", error);
+            }
+        }
+        else if (IsAdobePhotoCollection(destination.selectedItem))
+        {
+            AdobePhotoCollection *selectedPhotoCollection = destination.selectedItem;
+            
+            // Upload assets to selected photo collection.
+            [AdobePhotoAsset create:assetName
+                         collection:selectedPhotoCollection
+                           dataPath:assetURL
+                        contentType:kAdobeMimeTypePNG
+                      progressBlock:nil
+                       successBlock:^(AdobePhotoAsset *asset)
+             {
+                 NSLog(@"Upload success: %@", assetName);
+             }
+                  cancellationBlock:nil
+                         errorBlock:^(NSError *error)
+             {
+                 NSLog(@"Upload failed: %@", error);
+             }];
+        }
+        else if (IsAdobePhotoCatalog(destination.selectedItem))
+        {
+            AdobePhotoCatalog *selectedPhotoCatalog = destination.selectedItem;
+            
+            // Upload assets to selelcted photo catalog.
+            [AdobePhotoAsset create:assetName
+                            catalog:selectedPhotoCatalog
+                           dataPath:assetURL
+                        contentType:kAdobeMimeTypePNG
+                      progressBlock:nil
+                       successBlock:^(AdobePhotoAsset *asset)
+             {
+                 NSLog(@"Upload success: %@", assetName);
+             }
+                  cancellationBlock:nil
+                         errorBlock:^(NSError *error)
+             {
+                 NSLog(@"Upload failed: %@", error);
+             }];
+        }
+    }
+    
+    // Uploading to libraries, then perform sync.
+    if ([destination.selectedItem isKindOfClass:[AdobeLibraryComposite class]])
+    {
+        // Perform sync so that the added assets are uploaded & a delegate callback is received on sync complete.
+        AdobeLibraryManager *libMgr = [AdobeLibraryManager sharedInstance];
+        [libMgr sync];
+    }
+    
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Uploading Assets"
+                                                                             message:message
```

Comment:
Small nitpick: Maybe add a little more information to `message` to show what is going on since we fire and forget the upload request.